### ISSUE TITLE
POST to api/v2/inventories/<ID>/inventory_sources/ with id and name gets 201 created (should be 400)

### ISSUE THREAD

**Issue by kdelee** [ansible/awx#6309]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

I was trying to associate an inventory script with an inventory as an inventory source. When I did a POST to api/v2/inventories/<ID>/inventory_sources/ with id and name I got a 201 created (should be 400) and an inventory source object back. I then launched an inventory update and the following traceback was produced:

```
Traceback (most recent call last): File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/tasks.py", line 1308, in run args = self.build_args(self.instance, private_data_dir, passwords) File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/tasks.py", line 2472, in build_args if getattr(settings, '%s_EXCLUDE_EMPTY_GROUPS' % src.upper()): File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/conf/settings.py", line 511, in __getattr_without_cache__ return getattr(self._wrapped, name) File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/conf/settings.py", line 418, in __getattr__ value = self._get_default(name) File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/conf/settings.py", line 405, in _get_default return getattr(self.default_settings, name) AttributeError: 'Settings' object has no attribute '_EXCLUDE_EMPTY_GROUPS'
```

Seems like the bug is that we are not requiring the `source` perhaps among other things and ignoring `id`

**Comment by ryanpetrello** [comment]
Yep, I can reproduce this.  Crazy that we've never noticed this before now cc @AlanCoding.

**Comment by kdelee** [comment]
OK, so you fixed a related bug....but maybe not the same bug I was thinking of

```
akit --base-url=https://mytower.com -c config/credentials.yml -p config/projects.yml 
Python 3.7.5 (default, Dec 15 2019, 17:54:26) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.10.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: org = v2.organizations.create()        

In [2]: inv = v2.inventory.create(organization=org)                                            

In [3]: inv_script = v2.inventory_scripts.create(organization=org)                             

In [4]: what = inv.related.inventory_sources.post({'id': inv_script.id, 'name': 'foo1'})       # this is what I thought was a bug because this does not associated the script w/ the inventory as a source

In [5]: what.update()                 # this is the bug you fixed         
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-5-56ba97bde1e9> in <module>
----> 1 what.update()

~/sfw/ansible/awx/awxkit/awxkit/api/pages/inventory.py in update(self)
    589         assert update_pg.can_update, \
    590             "The specified inventory_source (id:%s) is not able to update (can_update:%s)" % \
--> 591             (self.id, update_pg.can_update)
    592 
    593         # start the inventory_update

AssertionError: The specified inventory_source (id:13) is not able to update (can_update:False)

```

**Comment by kdelee** [comment]
@ryanpetrello commented that script up there...let me know what you think

**Comment by ryanpetrello** [comment]
So, @kdelee initially when I looked at this, I thought about making it so that you *couldn't create* an Inventory Source without a `source` set, but I think that's the wrong approach, because _technically_ `source=''` **is** (or at least, at some point, *was*) a valid thing, as it's equivalent to a "Manual" inventory source.  Manual inventory sources are a thing that I _think_ are from a time long past, but I'm not certain that there aren't users out there still relying on this for something.  So to me, it made more sense to _allow_ an empty source value, but to prevent _Tower_ from _launching_ syncs based on those sources.

**Comment by ryanpetrello** [comment]
I may need @AlanCoding, @matburt and/or @wenottingham to `<wave hands>` and explain the history and usage of manual inventory sources a bit more, as I'll admit I don't know a ton about them (or even if they're still especially relevant).

(see: https://github.com/ansible/awx/pull/6484#issuecomment-606189970)

**Comment by kdelee** [comment]
Becoming convinced....as it is app wide behavior to ignore unused arguments (in this case, `id`)....and that for "historic" reasons that `source` has a default of `''` and that this means a manual inventory source.....acceptable.

Will wait on the history but leaning towards calling this verified.

Wonder if a "manual" inventory source is when you do `awx-manage inventory_import --inventory-name HelloWorld --source /tmp/myinvfile.ini`

**Comment by AlanCoding** [comment]
https://github.com/ansible/awx/issues/5963#issuecomment-591214527

https://github.com/ansible/awx/blob/cfb633e8a643b0190fa07b6204b339a1d336cbb3/awx/main/migrations/_inventory_source.py#L8-L17

----

To me, the problem is that it was only half-way removed when it should have been fully removed, and because of that, no one can figure out if it's safe to remove the other half.

Removal requires a schema change, which is plenty worth it to me, because it that avoids any more time spent on it. Delete any remaining manual sources (for the second time).

For coherency and completeness, I suggest that the source choice is removed along with this field.

https://github.com/ansible/awx/blob/831bf9124f6b3a8cb0707d4651141bb5db13caab/awx/main/models/inventory.py#L1240

I expect no fallout from unexpected use cases after doing so. This is API v1 stuff.

I'm still fine if you want to punt on it, because it's better to take our time and do a more complete set of removals.

**Comment by ryanpetrello** [comment]
@kdelee I've opened a new PR that cleans up these old sources, and prevents them from being created entirely.

**Comment by kdelee** [comment]
Ok, this works for me, now you cannot even make the manual inventory source which was what `source` being equal to `""` used to mean.
